websub-ballerina/src/websub/tests/unit-tests/test_discovery_response_parsing.bal

```diff
@@ -0,0 +1,226 @@
+// Copyright (c) 2018 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerina/test;
+
+final string HUB_ONE = "https://hub.ballerina.com";
+final string HUB_TWO = "https://two.hub.ballerina.com";
+final string HUB_THREE = "https://three.hub.ballerina.com";
+
+final string TOPIC_ONE = "https://topic.ballerina.com";
+
+@test:Config {
+}
+function testTopicAndSingleHubAsSingleLinkHeader() {
+    http:Response response = new;
+    response.addHeader("Link", "<" + HUB_ONE + ">; rel=\"hub\", <" + TOPIC_ONE + ">; rel=\"self\"");
+    var results = extractTopicAndHubUrls(response);
+    if (results is [string, string[]]) {
+        string topic = results[0];
+        string[] hubs = results[1];
+        test:assertEquals(topic, TOPIC_ONE, msg = "incorrect topic extraction from discovery response");
+        test:assertEquals(hubs.length(), 1, msg = "incorrect no. of hubs extracted from discovery response");
+        test:assertEquals(hubs[0], HUB_ONE, msg = "incorrect hub extraction from discovery response");
+    } else {
+        string? errMsg = results.message();
+        test:assertFail(msg = errMsg is string ? errMsg : "error in extractTopicAndHubUrls");
```

Comment:
The error.message() returns a string with the recent error change. So shall we change this assertion as below? We may need to fix other places as well
```suggestion
        test:assertFail(msg = results.message());
```

Reply:
Fixed in here https://github.com/ballerina-platform/module-ballerina-websub/pull/1/commits/7defc19195f199d3899556e82f9a35eb1ebf4bd5